nipype/utils/filemanip.py

```diff
@@ -690,32 +701,50 @@ def load_json(filename):
 
     """
 
-    with open(filename, 'r') as fp:
+    with open(filename, "r") as fp:
         data = json.load(fp)
     return data
 
 
 def loadcrash(infile, *args):
-    if infile.endswith('pkl') or infile.endswith('pklz'):
+    if infile.endswith("pkl") or infile.endswith("pklz"):
         return loadpkl(infile)
     else:
-        raise ValueError('Only pickled crashfiles are supported')
+        raise ValueError("Only pickled crashfiles are supported")
 
 
 def loadpkl(infile):
     """Load a zipped or plain cPickled file."""
     infile = Path(infile)
-    fmlogger.debug('Loading pkl: %s', infile)
-    pklopen = gzip.open if infile.suffix == '.pklz' else open
+    fmlogger.debug("Loading pkl: %s", infile)
+    pklopen = gzip.open if infile.suffix == ".pklz" else open
 
-    with SoftFileLock('%s.lock' % infile):
-        with pklopen(str(infile), 'rb') as pkl_file:
-            pkl_contents = pkl_file.read()
+    t = time()
+    timeout = float(config.get("execution", "job_finished_timeout"))
+    timed_out = True
+    while (time() - t) < timeout:
+        try:
+            glob(str(infile)).pop()
+            timed_out = False
+            break
+        except Exception as e:
+            fmlogger.debug(e)
```

Comment:
Is there something special about `glob`, as opposed to, e.g.,

```Python
if infile.exists():
    timed_out = False
    break
fmlogger.debug("'{}' missing; waiting 2s".format(infile))
```

This feels more straightforward, so if `glob` is doing something useful beyond an `exists()` check, a comment would be useful.

Reply:
updated the code - that code is old, and i don't remember why we used glob.